### PR TITLE
aarch64: add vcpu configuration

### PR DESCRIPTION
**Pull request by dianpopa**
This sets up the aarch64 specific registers.

Signed-off-by: Diana Popa <dpopa@amazon.com>

Issue #, if available:

Description of changes:

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: arch/src/aarch64/layout.rs [hunk 1]
```diff
@@ -59,6 +59,9 @@ pub const CMDLINE_START: usize = 0x0;
 /// Kernel command line start address maximum size.
 pub const CMDLINE_MAX_SIZE: usize = 0x0;
 
+/// Maximum size of the device tree blob as specified in https://www.kernel.org/doc/Documentation/arm64/booting.txt.
+pub const FDT_MAX_SIZE: usize = 0x200000;
```
Comment (alexandruag) [inline arch/src/aarch64/layout.rs]: Looks a bit nicer as `0x20_0000`.

File: vmm/src/vstate.rs [hunk 2]
```diff
@@ -467,6 +505,40 @@ mod tests {
...
+            0,
+            &vm,
+            devices::Bus::new(),
+            devices::Bus::new(),
+            super::super::TimestampUs::default(),
+        )
+        .unwrap();
+
+        let vm_config = VmConfig::default();
+        assert!(vcpu.configure(&vm_config, GuestAddress(0), &vm).is_ok());
```
Comment (alexandruag) [inline vmm/src/vstate.rs]: Can you also test here that the features which you set via `self.fd.vcpu_init(&kvi)?;` in `configure` are what you'd expect? I guess this also applies to the next `assert`.

File: vmm/src/vstate.rs [hunk 3]
```diff
@@ -278,6 +281,41 @@ impl Vcpu {
         Ok(())
     }
 
+    #[cfg(target_arch = "aarch64")]
+    /// Configures an aarch64 specific vcpu and should be called once per vcpu from the vcpu's thread.
```
Comment (acatangiu) [inline vmm/src/vstate.rs]: nit: it is not actually necessary to call this from the vcpu's thread, and we actually don't do that.

File: arch/src/aarch64/regs.rs [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+const PSR_I_BIT: u64 = 0x00000080;
+const PSR_A_BIT: u64 = 0x00000100;
+const PSR_D_BIT: u64 = 0x00000200;
+// Taken from arch/arm64/kvm/inject_fault.c.
+const PSTATE_FAULT_BITS_64: u64 = (PSR_MODE_EL1h | PSR_A_BIT | PSR_F_BIT | PSR_I_BIT | PSR_D_BIT);
+
+// Following are macros that help with getting the ID of a aarch64 core register.
+// The core register are represented by the user_pt_regs structure. Look for it in
+// arch/arm64/include/uapi/asm/ptrace.h.
+macro_rules! offset__of {
```
[the remark this answers is not in the document]
Comment (dianpopa) [reply arch/src/aarch64/regs.rs]: Did it, take a look and let me know.

**Pull request merged**